magento-research/pwa-studio
# MiniCart Checkout Button

## Description


This PR updates the measurement the MiniCart uses to compute its `height` to use `window.innerHeight` instead of `100vh` as described by https://css-tricks.com/the-trick-to-viewport-units-on-mobile/. 

As described in the attached issue #1071, `100vh` often doesn't account for browser address bars or other frames.

This PR takes advantage of the new [useWindowSize](https://github.com/magento-research/pwa-studio/blob/develop/packages/peregrine/src/hooks/useWindowSize.js) hook to update this value even after resize events occur.

## Related Issue




Closes #1071 .

## Verification Steps

1. Add an item to your cart
1. Open the MiniCart (right drawer)
1. See that the Checkout button is fully visible
1. Resize the window (or change orientation of device) and ensure Checkout button is fully visible
1. The payments form depended a bit on these CSS variables, so ensure that looks good too

Run the verification steps above on a variety of phones.

## How Have YOU Tested this?




1. Tested using iPhone XR on iOS simulator
1. `yarn test`.

## Screenshots / Screen Captures (if appropriate):

### Before Fix
![image](https://user-images.githubusercontent.com/13182778/58664408-2e2f6580-82f4-11e9-8272-37d17557e266.png)

### After Fix
![Screen Shot 2019-05-30 at 3 35 28 PM](https://user-images.githubusercontent.com/13182778/58664336-0f30d380-82f4-11e9-8c4e-2ae6098366bd.png)

## Proposed Labels for Change Type/Package




## Checklist:



## Note

This PR is based on #1266 so that it can take advantage of the `useWindowSize` hook.

As such, it contains commits from #1266. The only code change introduced by this PR is https://github.com/magento-research/pwa-studio/pull/1283/commits/b7cabd5ff27dcd74dae7103601a9a3a0da27c02c.

I have added the `hold` label because this PR should not be merged before #1266 .
